Formula/r/rust.rb

```diff
@@ -222,6 +222,22 @@ def post_install
     end
   end
 
+  def caveats
+    <<~EOS
+      If you have installed rustup, the toolchain provided by this package may
+      introduce conflicts.
+
+      To address this issue, please make sure that your rustup installation comes
+      BEFORE this package in $PATH. This is usually done by modifying your shell's
+      configuration (e.g. .bashrc/.zshrc) so that Homebrew's `eval $(brew shellenv)`
+      line is evaluated BEFORE rustup's `. $HOME/.cargo/env` line (the actual syntax
+      might vary).
+
+      You may also want to link this toolchain with `rustup` under the name `system`:
+        rustup toolchain link system "$(brew --prefix rust)"
```

Comment:
When would you want to do this/not want to do this? Is this instead of the stuff above or as well?

Reply:
I could add an extra line saying `` this will enable commands like `rustc +system --version` ``.

Reply:
Again, sorry, not sure I understand! Can you ELI5 this too?

Reply:
@MikeMcQuaid Sure! As explained in https://github.com/Homebrew/homebrew-core/pull/218316#discussion_r2031485512, rustup expects the shims to shadow everything else, so how can we still use the `brew`-installed `rust` toolchain which should work perfectly alright? `rustup toolchain link` resolves this problem by adopting the said toolchain under a user-specified name (here `system`). So now if everything goes to plan, you can call the said toolchain's `rustc` with `rustc +system`:

```console
> rustc +system --version
rustc 1.85.1 (4eb161250 2025-03-15) (Homebrew)
```

Reply:
> Is this a system-wide "link" (that we could run automatically for the user) or a user-specific one?

@MikeMcQuaid The scope of this link is the same as that of the `rustup` installation, so in the case of `brew`-managed `rustup`, it is identical to that of all formulae. For example, if this is a system-wide `brew` installation, then both `rustup` and `rust` have the system scope, so `rustup toolchain link system` will have no incoherence in that regard.

Reply:
> Thanks! Is this a system-wide "link" (that we could run automatically for the user) or a user-specific one?

@MikeMcQuaid If you are talking about the scope of the "link" configuration, then it's limited to the current user by default, because we store that config under `$RUSTUP_HOME` which defaults to `~/.rustup/`. Setting that value to other stuff would be possible but looks a bit weird to me...

Reply:
Ok, great, thanks. Feels a bit like `brew rbenv-sync`, if we ever wanted to integrate like that 🤔